### PR TITLE
Round on a column was always adding a warning

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Only warn if outside allowed range.
- Added `is_infinite` to In-Memory column.
- Allow integer value type for `is_nan` and `is_infinite`.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Any.enso [hunk 1]
```diff
@@ -300,6 +289,16 @@ type Any
...
+    ## PRIVATE
+
+       Executes the provided handler on a dataflow error, or executes as
+       identity on a non-error value.
+
+       Arguments:
+       - handler: The function to call on this if it is an error value.
+    catch_primitive : (Error -> Any) -> Any
+    catch_primitive self handler = @Builtin_Method "Any.catch_primitive"
+
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Any.enso:301]: Curious - why is this moved?

Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Any.enso:301]: put it next to catch - no real reason.

File: std-bits/table/src/main/java/org/enso/table/data/column/storage/numeric/AbstractLongStorage.java [hunk 2]
```diff
@@ -279,6 +279,22 @@ public BoolStorage run(AbstractLongStorage storage) {
                 return new BoolStorage(storage.getIsMissing(), new BitSet(), storage.size(), false);
               }
             })
+        .add(
+            new UnaryMapOperation<>(Storage.Maps.IS_NAN) {
+              @Override
+              public BoolStorage run(AbstractLongStorage storage) {
+                BitSet isNaN = new BitSet(storage.size());
```
Comment (radeusgd) [inline std-bits/table/src/main/java/org/enso/table/data/column/storage/numeric/AbstractLongStorage.java]:
```suggestion
                BitSet isNaN = new BitSet();
```

I think this will be valid too - IIRC the BitSet will happily answer `false` if `get` index is higher than its size. But then this will avoid allocating an array of size N - so we should get a O(1) storage for this case. Worth trying IMO :)

**Review by radeusgd — COMMENTED** [review]
Looks good, but please let's add some tests for `is_nan` and `is_infinite` for Integer columns - just to verify that everything is correctly wired-up

File: distribution/lib/Standard/Base/0.0.0-dev/src/Any.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Any.enso:301]: Ok, that sounds like a good reason, I just was wondering if there's anything deeper here :)

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Column.enso [hunk 3]
```diff
@@ -732,32 +732,32 @@ type Column
...
-        nothings = self.const nothing_with_problem . cast target_value_type
-        result = within_range.iif column nothings
-        result.rename column.name
+        within_range = if self.value_type.is_floating_point.not then self.between min max else
+            self.is_nan || self.is_infinite || self.between min max
+
+        any_issues = within_range.to_vector . any (== False)
+        if any_issues.not then action self else
+            warning = Illegal_Argument.Error <| "Error: `round` can only accept values between " + min.to_text + " and " + max.to_text + " (inclusive)"
+            result = action (within_range.iif self Nothing)
```
Comment (GregoryTravis) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Data/Column.enso:759]: Since we're already converting to vector and traversing the column, we could just build up the result with results and Nothings at the same time, to save an extra pass.

**Review by GregoryTravis — APPROVED** [review]

File: test/Table_Tests/src/In_Memory/Column_Spec.enso [hunk 4]
```diff
@@ -117,6 +117,9 @@ spec =
             Column.from_vector "foo" [12, 24, 25, 29] . round 0 . value_type . should_equal Value_Type.Integer
             Column.from_vector "foo" [12, 24, 25, 29] . round -1 . value_type . should_equal Value_Type.Integer
 
+        Test.specify "rounding should not attach a warning by default" <|
+            Column.from_vector "foo" [12, 24, 25, 29] . round 1 . has_warnings . should_be_false
```
Comment (radeusgd) [inline test/Table_Tests/src/In_Memory/Column_Spec.enso]: We have a more specialized check for this, probably worth using it
```suggestion
            Problems.assume_no_problems <| Column.from_vector "foo" [12, 24, 25, 29] . round 1
```

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Column.enso (hunk 3 above)
Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Data/Column.enso:759]: Sounds good but is that possible without changing more code? I tihnk then we cannot easily use the vectorized operations

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data/Column.enso [hunk 5]
```diff
@@ -732,32 +732,32 @@ type Column
...
-        nothing_with_problem = Warning.attach warning Nothing
-        ## TODO: This should be cast to NULL when that type is available,
-           see https://github.com/enso-org/enso/issues/6281.
-        nothings = self.const nothing_with_problem . cast target_value_type
-        result = within_range.iif column nothings
-        result.rename column.name
+        within_range = if self.value_type.is_floating_point.not then self.between min max else
+            self.is_nan || self.is_infinite || self.between min max
+
+        any_issues = within_range.to_vector . any (== False)
```
Comment (radeusgd) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Data/Column.enso:756]: Btw. do we have to do the `Vector.any` here? I think we should be able to do: `self.to_table.aggregate (Aggregate_Column.Maximum 0) != False` which should be equivalent but will use the vectorized Java ops.

**Review by radeusgd — APPROVED** [review]
Looks good. I'd consider using an `aggregate` instead of `Vector.any` to stay with more 'vectorized' approach for the operations.
